lib/Macros/Sources/SwiftMacros/DebugDescriptionMacro.swift

```diff
@@ -0,0 +1,312 @@
+    decl.as(StructDeclSyntax.self)?.genericParameterClause?.parameters ??
+    decl.as(ClassDeclSyntax.self)?.genericParameterClause?.parameters ??
+    decl.as(EnumDeclSyntax.self)?.genericParameterClause?.parameters
+}
+
+private func bindingName(_ binding: PatternBindingSyntax) -> String? {
+    binding.pattern.as(IdentifierPatternSyntax.self)?.identifier.text
+}
+
+private func isComputedProperty(_ binding: PatternBindingSyntax) -> Bool {
+    guard let accessors = binding.accessorBlock?.accessors else {
+        // No accessor block, not computed.
+        return false
+    }
+
+    switch accessors {
+    case .accessors(let accessors):
+        return accessors.contains { $0.accessorSpecifier.tokenKind == .keyword(.get) }
+    case .getter:
+        return true
+    @unknown default:
+        return false
+    }
+}
+
+// Enumerate a MemberAccess expression to produce DeclReferences in left to right order.
+// `a.b.c` will result in callbacks for each DeclReference `a`, `b`, and then `c`.
+private func enumerateMembers(_ memberAccess: MemberAccessExprSyntax, _ action: (DeclReferenceExprSyntax) -> Void) {
+    if let baseMember = memberAccess.base?.as(MemberAccessExprSyntax.self) {
+        enumerateMembers(baseMember, action)
+    } else if let baseDecl = memberAccess.base?.as(DeclReferenceExprSyntax.self) {
+        action(baseDecl)
+    }
+    action(memberAccess.declName)
+}
+
+// MARK: - Encoding
+
+/// Construct an LLDB type summary record.
+///
+/// The record is serializeed as a tuple of `UInt8` bytes.
+///
+/// The record contains the following:
+///   * Version number of the record format
+///   * The size of the record (encoded as ULEB)
+///   * The type identifier, which is either a type name, or for generic types a type regex
+///   * The description string converted to an LLDB summary string
+///
+/// The strings (type identifier and summary) are encoded with both a length prefix (also ULEB)
+/// and with a null terminator.
+private func encodeTypeSummaryRecord(_ typeIdentifier: String, _ summaryString: String) -> String {
+    let encodedType = encodeString(typeIdentifier)
+    let encodedSummary = encodeString(summaryString)
+    let recordSize = UInt(encodedType.count + encodedSummary.count)
+    return """
+    /* version */ 1 as UInt8,
+    /* record size */ \(literalBytes(encodeULEB(recordSize))),
+    /* "\(typeIdentifier)" */ \(literalBytes(encodedType)),
+    /* "\(summaryString)" */ \(literalBytes(encodedSummary))
+    """
+}
+
+/// Generate a _partial_ Swift literal from the given bytes. It is partial in that must be embedded
+/// into some other syntax, specifically as a tuple.
+private func literalBytes(_ bytes: [UInt8]) -> String {
+    bytes.map({ "\($0) as UInt8" }).joined(separator: ", ")
+}
+
+/// Encode a string into UTF8 bytes, prefixed by a ULEB length, and suffixed by the null terminator.
+private func encodeString(_ string: String) -> [UInt8] {
```

Comment:
Similarly here, I would make all these private functions that take a single parameter private extensions.